lib/dealership.rb

```diff
@@ -0,0 +1,78 @@
+class Dealership
+  attr_reader :name,
+              :address,
+              :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+  end
+
+  def add_car(car_instance)
+    @inventory << car_instance
+  end
+
+  def inventory_count
+    @inventory.length
+  end
+
+  def cars_by_make(make_of_car)
+    @inventory.find_all do |car|
+      car.make == make_of_car
+    end
+  end
+
+  def total_value
+    @inventory.sum do |car|
+      car.total_cost
+    end
+  end
+
+  def details
+    details_hash ={
+      "total_value" => total_value,
+      "address" => @address
+    }
+    details_hash
+  end
+
+  def average_price_of_car
+    average_price = (total_value / @inventory.length).to_s
+    first_two = average_price.slice(0..1)
+    last_two = average_price.slice(2..4)
+
+    "#{first_two},#{last_two}"
+
+    ## I FOUND THIS BELOW METHOD ONLINE THAT IS REALLY DYNAMIC. I DO NOT UNDERSTAND IT BUT WOULD LIKE TO LEAVE IT HERE SO I CAN TRY TO UNDERSTAND LATER. THIS IS A HARD METHOD. PLEASE HELP ME UNDERSTAND THIS SOMETIME
+
+    # number_string_format = (total_value / @inventory.length).to_s.chars.reverse.each_slice(3)
+    # number_string_format.map(&:join).join(',').reverse
```

Comment:
I can explain it a bit breaking down by each step:

(total_value/@inventory.length) = average price for ex. 42000

42000.to_s -> "42000"
"42000".char -> ['4', '2', '0', '0', '0']

Now we reverse the array because we know we need a comma every 3 digits when writing American English style numbers. Essentially we want to count from the back of the above array. The easiest way to do that is to first reverse the array:

['4', '2', '0', '0', '0'].reverse = ['0', '0', '0', '2', '4']

now we can group by the 3 digits we need

['0', '0', '0', '2', '4'].each_slice(3) = creates an enumerator when the value your iterating through is an array of up to 3 items [ [0,0,0], [2,4] ] 

Now we wish to put commas between those groupings of digits:

[ [0,0,0], [2,4] ] .map(&:join) = ["000", "24"]

 ["000", "24"].join(',') = "000,24"
 
then reversing that entire string will give us "42,000"

Definitely an advanced topic and good on you for putting in comments that you found it but wasn't really sure why it worked the way it did.